### ISSUE TITLE
Put notations in separate file

### ISSUE THREAD

**Issue by benediktahrens** [UniMath/UniMath#913]
Some notations are not always useful, and even harmful; eg., the notation for transportf and transportb hides the information of the dependent type family.
These not-so-innocent notations should be defined in a separate file, that can be imported by those who want them.

**Comment by DanGrayson** [comment]
I agree.

Alternatively, what if we just make it "only parsing"?  Then it will never be a problem, and we only have to change one line.

**Comment by siddharthist** [comment]
I agree for transport especially, I am continually frustrated by this.

**Comment by benediktahrens** [comment]
@DanGrayson : thanks a lot for #914 , that is a big improvement.
My goal, however, is to have a more general discussion on how to handle notation. I am thus reopening this issue.

**Comment by DanGrayson** [comment]
Here's a related problem with notations I'd like to add to this discussion: it is currently impossible to load all of the UniMath files into Coq simultaneously, because of global notations in various files that conflict with each other.  To track this down I'm adding a rule to the make file that makes a file `UniMath/All.v` and will get it compiled.

The most convenient way to prevent notation conflicts is for notations to be reserved in `Foundations/Init.v`, where they are given a parsing precedence and an associativity, and for all the other files not to specify those attributes of the notation.  That way, all the files will automatically agree.

The most inconvenient way to prevent notation conflicts is to wait until travis tells us there is one, via an error message while compiling `UniMath/All.v`.

Exempting local notations from this practice is not workable, because eventually some other developer will want to use the same notation globally, will add it to `Foundations/Init.v`, and then the conflict will arise anyway.
